### PR TITLE
Initialize sensitive terms list

### PR DESCRIPTION
**Pull request by zackkrida**
> **Warning**
>
> This PR contains text of a sensitive nature. Examples of potentially sensitive text include but are not limited to text of a sexual, biological, political, religious, violent, or racist nature.

This PR initializes the sensitive terms list with the combined 1913 terms from two public lists:

- https://github.com/LDNOOBW/List-of-Dirty-Naughty-Obscene-and-Otherwise-Bad-Words
- https://github.com/surge-ai/profanity

I _have not_ yet removed or recommended for removal any specific terms which may be unsuitable for this project.

## Resources for evaluating terms

- WordPress's page on [etiquette](https://wordpress.org/about/etiquette/) in the project.
- WordPress's [code of conduct](https://make.wordpress.org/community/handbook/wordcamp-organizer/planning-details/code-of-conduct/) for in-person events.
- YouTube's [community guidelines](https://www.youtube.com/howyoutubeworks/policies/community-guidelines/) offers a decent list of categories for unacceptable types of content. For example, their [thumbnail policy](https://support.google.com/youtube/answer/9229980?hl=en&ref_topic=9282679#zippy=%2Cage-restricted-thumbnails-and-thumbnail-removal) includes some information on types of sexual and violent content that either can't be used or can be used behind a flag comparable to ours.

### CODE REVIEW

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
I think there are some terms we can remove and we should add the canonical forms to make sure they're actually included. Additionally, we can set everything to lowercase and remove any duplicates resulting just from capitalisation differences.

Some ideas for future improvements to this feature that would be useful:

- Quote terms _here_ instead of during query building time. This would allow us to combine terms that together more confidently match sensitive documents.
- Have a secondary list of sensitive terms that we only use for displaying when they appear on results that are also `mature=true`. That way we could potentially disambiguate really heinous terms that have common innocuous usage and still call out sensitive text on those results in addition to the mature flag on them. This would allow the sensitive text warning to appear on results that have some terms that we don't blanket mark as sensitive.

File: sensitive-terms.txt
Comment (obulat) [reply sensitive-terms.txt:161]: I agree with your suggestion, @sarayourfriend . I don't think we should include "bbc" as there is so much more content for the media company. I tried about 8-10 pages, and couldn't find more than 2 sensitive images (I haven't checked all the titles, though).

**Comment by sarayourfriend** [comment]
Another idea occurred to me. For things like the word "sperm", if we want to mark some results as sensitive but leave "sperm whale" untouched, we could use a query string rather than multi-match. We'd need to add a prefix to such lines so the ingestion server can find them and use the right type of query. Maybe something like DuckDuckGo's "bangs" could work:

```
!qs:sperm -whale
```

That would get transformed to:

```json
"query": {
  "must_not": [
    // ...
    { "query_string": { "query": "sperm -whale", "fields": ["title", "description", "tags.name"] } }
  ]
}
```

The reason this works (confused me for a bit, but I confirmed it in staging, details below), is that it will include anything that matches "sperm" that doesn't also match "whale". That prevents the whale from being marked as sensitive.

I believe there are multiple terms we might be able to employ this for, including "peter".

The default would be to just use a multi-match query. Given the vast majority are going to just be quoted single terms/phrases, we could make it so that lines without bangs are always quoted as-is, and the bangs instruct the ingestion server to do something different. That might make certain terms easier to "modify" so that they only apply to truly problematic instances.

I tested this in staging using a filtered alias and a compound boolean query:

**`POST /image-sensitive-text-2023-03-23/_alias/sensitive-text-match-testing__sara__no-parties-without-hats`:**
```json
{
	"filter": {
		"bool": {
			"must_not": [
				{
					"query_string": {
						"query": "party -hat"
					}
				}
			]
		}
	}
}
```

This makes it so that "party" is only allowed in documents if "hat" appears alongside it, but it is not allowed by itself. We can query the alias to prove that:

**`POST /sensitive-text-match-testing__sara__no-parties-without-hats/_search`:**

```json
{
	"query": {
		"bool": {
			"must": {
				"query_string": {
					"query": "cat AND party"
				}
			},
			"must_not": [
				{
					"query_string": {
						"query": "hat"
					}
				}
			]
		}
	}
}
```

That returns zero hits because there simply are no parties without hats. If we query the unfiltered origin index, we get plenty of results for "cats AND party" without hat anywhere in the document. If you remove the `must_not` clause and query the filtered alias again, you will get lots of results for cats at parties with hats.

So anyway, all of that to say: there are workarounds for the edge cases we might find. I don't even think they'd be a huge lift to accommodate on the ingestion server side of things.

**Comment by sarayourfriend** [comment]
I keep thinking of more use cases for the idea in my previous comment. Marking anything that mentions the scientific terms for human body parts (vagina, penis, breast, etc) feels... strange to me. While I want people to be able to opt into things, I also don't want to reproduce the kinds of ideas that lead to American schools banning biology text books that speak about the human anatomy. I think we could improve the situation for some of those terms by using the qs syntax. For example, we could only apply the sensitivity check to specific providers by doing `!qs:breast provider:flickr` (and so on). We could even use that to exclude specific creators on a particular provider. Say we know someone publishes high quality images useful for biology education, we could do: `!qs:breast provider:flickr -creator:AwesomeEducator`[^w]. That would hide everything mentioning breasts on flickr except from the creator "AwesomeEducator".

[^w]: This might not be the exact syntax, I'm not looking at the query syntax guide right now, the `-` might need to go inside the field query rather than in front of the field name; in any case, it coveys the idea.

It also occurs to me that we could start this list with things that are less ambiguous and add the more ambiguous ones later once we understand more about how to target specific usages.

Scoping by provider seems like a pretty good way of being to include _more_ terms without accidentally including high quality, non-derogatory/sexualised/etc usages of some of those terms.

**Comment by zackkrida** [comment]
@sarayourfriend I'm going to start by updating the document to remove any words that do not belong based on your suggestions. Regarding the idea to use a query string rather than multi-match, I think that's a phenomenal idea. It suggests to me moving this sensitive terms list to a format like json5 or yaml though rather than coming up with a bespoke string syntax. That might be more flexible for other exceptions or query modifications we might want to make in the future. 

I think I'd want to write a mini-implementation plan before switching to that. My inclination here would be to do as you mentioned here:

> It also occurs to me that we could start this list with things that are less ambiguous and add the more ambiguous ones later once we understand more about how to target specific usages.

Namely, to start with a text list of obviously-explicit terms and then expand in the future to a different data format and thoughtful targeting of ambiguous terms. As the project lead do you think that's an acceptable approach? I'm also content to explore this now, so I'll leave it up to you.

**Comment by zackkrida** [comment]
Normalized all to lowercase and removed likely-innocuous terms.

**Comment by sarayourfriend** [comment]
> Namely, to start with a text list of obviously-explicit terms and then expand in the future to a different data format and thoughtful targeting of ambiguous terms. As the project lead do you think that's an acceptable approach? I'm also content to explore this now, so I'll leave it up to you.

I need to think about this more, but I don't think it blocks this PR as it'll be useful either way to have this terms list available for use in the implementation of the filtered index. It should have an implementation plan, absolutely. But I would also like to consider how we should prioritise it. Should it be a "fast follow" to the filtered index work? Or, is this approach valuable enough that we would not consider the MVP without it? I'm inclined to say that if we stick to unambiguous terms for now, it is fine to do this as a quick "phase 2" of the filtered index project once the initial implementation is finished.

To me that has the advantages of giving us time to ensure that the basic idea is truly viable. We strongly suspect it is, but we can verify it by trying it first, and then adding further iterations on the filtered index creation later on.

**Comment by zackkrida** [comment]
> I'm inclined to say that if we stick to unambiguous terms for now, it is fine to do explore making filtered index creation queries more flexible as a quick "phase 2" of the filtered index project once the initial implementation is finished.
>
> To me that has the advantages of giving us time to ensure that the basic idea is truly viable. We strongly suspect it is, but we can verify it by trying it first, and then adding further iterations on the filtered index creation later on.

Agreed, thank you for thinking through this with me!

**Comment by zackkrida** [comment]
@sarayourfriend and @obulat I believe this is ready for re-review; thank you!

**Review by sarayourfriend — APPROVED** [review]
LGTM. We can iterate on this as needed.

File: sensitive-terms.txt
Comment (obulat) [inline sensitive-terms.txt:115]: Why don't we have `b!tch`, too, without a space?

File: sensitive-terms.txt
Comment (obulat) [inline sensitive-terms.txt:484]: I'm not sure about this one - most of the results that come with this search are not sensitive

**Review by obulat — APPROVED** [review]
Sorry for late review. Agree that this is a good start.

File: sensitive-terms.txt
Comment (zackkrida) [reply sensitive-terms.txt:115]: The space might be a typo, I'll fix. thanks!

**Pull request merged**
